drivers/meter/ade9153a/ade9153a.h

```diff
@@ -0,0 +1,1470 @@
+	/* Select the number of temperature readings to average*/
+	/* New temperature measurement every 1ms */
+	ADE9153A_TEMP_TIME_SAMPLES_1,
+	/* New temperature measurement every 256 ms */
+	ADE9153A_TEMP_TIME_SAMPLES_256,
+	/* New temperature measurement every 512 ms */
+	ADE9153A_TEMP_TIME_SAMPLES_512,
+	/* New temperature measurement every 1 sec */
+	ADE9153A_TEMP_TIME_SAMPLES_1024
+};
+
+/**
+* @enum ade9153a_ai_gain_e
+* @brief ADE9153A PGA gain for current CH A
+*/
+enum ade9153a_ai_gain_e {
+	/* PGA gain*/
+	/* Gain = 16 */
+	ADE9153A_AI_GAIN_16 = 2,
+	/* Gain = 24 */
+	ADE9153A_AI_GAIN_24 = 3,
+	/* Gain = 32 */
+	ADE9153A_AI_GAIN_32 = 4,
+	/* Gain = 38.4 */
+	ADE9153A_AI_GAIN_38_4 = 5
+};
+
+/**
+* @enum ade9153a_chip_stat_err_e
+* @brief ADE9153A chip status errors
+*/
+enum ade9153a_chip_stat_err_e {
+	/* Error 0 - SW/HW reset to clear */
+	ADE9153A_ERROR0,
+	/* Error 1 - SW/HW reset to clear */
+	ADE9153A_ERROR1,
+	/* Error 2 - SW/HW reset to clear */
+	ADE9153A_ERROR2,
+	/* Error 3 - SW/HW reset to clear */
+	ADE9153A_ERROR3,
+	/* UART Error 0 - UART reset to clear */
+	ADE9153A_UART_ERROR0,
+	/* UART Error 1 - UART reset to clear */
+	ADE9153A_UART_ERROR1,
+	/* UART Error 2 - UART reset to clear */
+	ADE9153A_UART_ERROR2,
+	/* UART interface reset detected */
+	ADE9153A_UART_RESET
+};
+
+/**
+* @enum ade9153a_phnoload_e
+* @brief Type of energy that is in no load condition
+*/
+enum ade9153a_phnoload_e {
+	/* No load */
+	/* Phase A fundamental reactive energy is in no load. */
+	AFVARNL = 1,
+	/* Phase A total apparent energy is in no load */
+	AVANL,
+	/* Phase A total active energy is in no load */
+	AWATTNL
+};
+
+/**
+* @enum dip_swell_irq_mode_en
+* @brief Dip swell irq mode
+*/
+enum dip_swell_irq_mode_en {
+	/* continuous */
+	CONTINUOUSE,
+	/* One interrupt when entering cond. another when exiting cond. */
+	ONE_INT,
+};
+
+/**
+* @struct ade9153a_init_param
+* @brief ADE9153a Device initialization parameters.
+*/
+struct ade9153a_init_param {
+	/* Device communication descriptor */
+	struct no_os_spi_init_param 	*spi_init;
+	/** GPIO RDY descriptor used to signal when ADC data is available */
+	struct no_os_gpio_init_param	*gpio_rdy;
+	/** GPIO RESET descriptor used to reset device (HW reset) */
+	struct no_os_gpio_init_param  	*gpio_reset;
+	/** GPIO ss descriptor used to config comms */
+	struct no_os_spi_init_param 	*gpio_ss;
+	/** GPIO sck descriptor used to config comms */
+	struct no_os_spi_init_param		*gpio_sck;
+	/** Enable SPI interface */
+	uint8_t							spi_en;
+	/** IRQ device descriptor used to handle interrupt routine for GPIO RDY */
+	struct no_os_irq_ctrl_desc 		*irq_ctrl;
+	/** operation of sensor */
+	uint8_t							ai_swap;
+	/** AI gain init value */
+	enum ade9153a_ai_gain_e			ai_pga_gain;
+	/** High pass filter corner freq init value */
+	enum ade9153a_hpf_crn_e			hpf_crn;
+	/** Energy accumulation freq select */
+	enum ade9153a_selfreq_e			freq;
+	/** Vlevel value */
+	uint32_t						vlevel;
```

Comment:
same comment about indentation applies to these structures, i usually avoid such indentations in structures and just use a single whitespace instead